planetscale/client.go

```diff
@@ -316,3 +361,13 @@ type Error struct {
 
 // Error returns the string representation of the error.
 func (e *Error) Error() string { return e.msg }
+
+// CursorPaginatedResponse provides a generic means of wrapping a paginated
+// response.
+type CursorPaginatedResponse[T any] struct {
+	Data        []T    `json:"data"`
+	HasNext     bool   `json:"has_next"`
+	HasPrev     bool   `json:"has_prev"`
+	CursorStart string `json:"cursor_start"`
+	CursorEnd   string `json:"cursor_end"`
```

Comment:
Would be nice to add a comment that this means the next cursor.

Reply:
I got you!